tests/testsuite/git.rs

```diff
@@ -4052,6 +4052,100 @@ src/lib.rs
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+    p.cargo("fetch")
+        .env("CARGO_NET_GIT_FETCH_WITH_CLI", "true")
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[UPDATING] git repository `https://github.com/rust-lang/bitflags.git`
+fatal: remote [ERROR] upload-pack: not our ref 11111b376b93484341c68fbca3ca110ae5cd2790
+[ERROR] failed to get `bitflags` as a dependency of package `foo v0.1.0 ([ROOT]/foo)`
+
+Caused by:
+  failed to load source for dependency `bitflags`
+
+Caused by:
+  Unable to update https://github.com/rust-lang/bitflags.git?rev=11111b376b93484341c68fbca3ca110ae5cd2790
+
+Caused by:
+  failed to clone into: [ROOT]/home/.cargo/git/db/bitflags-[HASH]
+
+Caused by:
+  revision 11111b376b93484341c68fbca3ca110ae5cd2790 not found
+
+Caused by:
+  process didn't exit successfully: `git fetch --no-tags --force --update-head-ok 'https://github.com/rust-lang/bitflags.git' '+11111b376b93484341c68fbca3ca110ae5cd2790:refs/commit/11111b376b93484341c68fbca3ca110ae5cd2790'` ([EXIT_STATUS]: 128)
+
+"#]])
+        .run();
+}
+
+#[cargo_test(public_network_test)]
+fn git_fetch_libgit2_error_message() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+                edition = "2015"
+
+                [dependencies]
+                bitflags = { git = "https://github.com/rust-lang/bitflags.git", rev="11111b376b93484341c68fbca3ca110ae5cd2790" }
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+    p.cargo("fetch")
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[UPDATING] git repository `https://github.com/rust-lang/bitflags.git`
+[WARNING] spurious network error (3 tries remaining): could not read from remote repository; class=Net (12); code=Eof (-20)
+[WARNING] spurious network error (2 tries remaining): could not read from remote repository; class=Net (12); code=Eof (-20)
+[WARNING] spurious network error (1 tries remaining): could not read from remote repository; class=Net (12); code=Eof (-20)
+[ERROR] failed to get `bitflags` as a dependency of package `foo v0.1.0 ([ROOT]/foo)`
+
+Caused by:
+  failed to load source for dependency `bitflags`
+
+Caused by:
+  Unable to update https://github.com/rust-lang/bitflags.git?rev=11111b376b93484341c68fbca3ca110ae5cd2790
+
+Caused by:
+  failed to clone into: [ROOT]/home/.cargo/git/db/bitflags-[HASH]
+
+Caused by:
+  revision 11111b376b93484341c68fbca3ca110ae5cd2790 not found
+
+Caused by:
+  network failure seems to have happened
+  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
+  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
+
+Caused by:
+  could not read from remote repository; class=Net (12); code=Eof (-20)
```

Comment:
It is interesting that the error message doesn't seem to be emitted on GitHub Actions machine, but only on my local machine.

If it still happens, instead of asserting the entire message, try omitting some parts we're not interested in with `...` (For the syntax, see https://doc.rust-lang.org/nightly/nightly-rustc/cargo_test_support/compare/index.html).


```suggestion
[UPDATING] git repository `https://github.com/rust-lang/bitflags.git`
...
[ERROR] failed to get `bitflags` as a dependency of package `foo v0.1.0 ([ROOT]/foo)`
Caused by:
  failed to load source for dependency `bitflags`
Caused by:
  Unable to update https://github.com/rust-lang/bitflags.git?rev=11111b376b93484341c68fbca3ca110ae5cd2790
Caused by:
  failed to clone into: [ROOT]/home/.cargo/git/db/bitflags-[HASH]
Caused by:
  revision 11111b376b93484341c68fbca3ca110ae5cd2790 not found
...
```

Reply:
For the context:

```
---- git::git_fetch_libgit2_error_message stdout ----
running `/Users/runner/work/cargo/cargo/target/debug/cargo fetch`
thread 'git::git_fetch_libgit2_error_message' panicked at tests/testsuite/git.rs:4163:10:

---- expected: tests/testsuite/git.rs:4135:27
++++ actual:   stderr
   1    1 | [UPDATING] git repository `[https://github.com/rust-lang/bitflags.git`](https://github.com/rust-lang/bitflags.git%60)
   2      - [WARNING] spurious network error (3 tries remaining): could not read from remote repository; class=Net (12); code=Eof (-20)
   3      - [WARNING] spurious network error (2 tries remaining): could not read from remote repository; class=Net (12); code=Eof (-20)
   4      - [WARNING] spurious network error (1 tries remaining): could not read from remote repository; class=Net (12); code=Eof (-20)
   5    2 | [ERROR] failed to get `bitflags` as a dependency of package `foo v0.1.0 ([ROOT]/foo)`
   6    3 | 
   7    4 | Caused by:
   8    5 |   failed to load source for dependency `bitflags`
   9    6 | 
          ⋮
  15   12 | 
  16   13 | Caused by:
  17   14 |   revision 11111b376b93484341c68fbca3ca110ae5cd2790 not found
  18   15 | 
  19   16 | Caused by:
  20      -   network failure seems to have happened
  21      -   if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  22      -   https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
       17 +   Could not decode server reply
  23   18 | 
  24   19 | Caused by:
  25      -   could not read from remote repository; class=Net (12); code=Eof (-20)
       20 +   Unknown or unsupported header: "ERR upload-pack: not our ref 11111b376b93484341c68fbca3ca110ae5cd2790"
       ```